cmd/ccboc/main.go

```diff
@@ -121,6 +122,83 @@ func getCalculationID(c *cli.Context) error {
 	return nil
 }
 
+func getCalculationResult(c *cli.Context) error {
+	teff := c.Float64("teff")
+	logG := c.Float64("logG")
+	path := c.String("results-download-path")
+
+	resp, err := http.Get(globalConfig.APIURL + "/calculations/results?teff=" + fmt.Sprintf("%0.1f", teff) + "&logg=" + fmt.Sprintf("%0.2f", logG))
+	if err != nil {
+		logrus.WithError(err).Fatal("Couldn't retrieve the data from headers.")
+	}
+
+	fileNameHeader := resp.Header.Get("Content-Disposition")
+	if fileNameHeader == "" {
+		logrus.WithError(err).Fatal("Unable to get the headers")
+	}
+	splitFileNameHeader := strings.Split(fileNameHeader, "=")
+
+	defaultPath, err := config.GetPathToCalculation(path, splitFileNameHeader[1])
+	if err != nil {
+		logrus.WithError(err).Fatal("Couldn't get the path user added to download the calculations.")
+	}
+
+	body, responseError, err := request("GET", globalConfig.APIURL+"/calculations/results?teff="+fmt.Sprintf("%0.1f", teff)+"&logg="+fmt.Sprintf("%0.2f", logG), bytes.NewBuffer(nil))
+	if err != nil {
+		logrus.WithError(err).Fatal("error while perfoming the http request")
+	} else if responseError != nil {
+		logrus.WithFields(logrus.Fields{"message": responseError.Message, "status_code": responseError.StatusCode}).Fatal("errors occurred")
+	}
+
+	err = config.CreateAndWriteFile(body, defaultPath)
+	if err != nil {
+		logrus.WithError(err).Fatal("Something went wrong while creating or writing to a file.")
+	}
+
+	logrus.Info("The calculations were downloaded into: ", defaultPath)
+
+	return nil
+}
+
+func getCalculationResultByID(c *cli.Context) error {
+	args := os.Args
+	calcID := args[len(args)-1]
+
+	path := c.String("results-download-path")
+
+	resp, err := http.Get(globalConfig.APIURL + "/calculations/results/" + calcID)
+	if err != nil {
+		logrus.WithError(err).Fatal("Couldn't retrieve the data from headers.")
+	}
+
+	fileNameHeader := resp.Header.Get("Content-Disposition")
+	if fileNameHeader == "" {
+		logrus.WithError(err).Fatal("Unable to get the headers")
```

Comment:
The log here is wrong. The err will be nil since there is no error. Also please be more detailed about which header we failed to get.